core/src/main/scala/org/apache/spark/deploy/history/HistoryPage.scala

```diff
@@ -68,6 +68,7 @@ private[history] class HistoryPage(parent: HistoryServer) extends WebUIPage("")
                 <div id="history-summary" class="row-fluid"></div> ++
                 <script src={UIUtils.prependBaseUri(request, "/static/historypage.js")}></script> ++
                 <script>setAppLimit({parent.maxApplications})</script>
+                <script>setDeferRender({parent.deferRender})</script>
```

Comment:
Where does this actually have an effect?

Reply:
Hi, this option can speed the history summary loading in front-end side.

See details in https://datatables.net/examples/ajax/defer_render.html.

Reply:
So this defers rendering until the table is displayed by expanding it?
Is there any downside -- why wouldn't we set it on all the expandable tables that are collapsed by default?
That is, I don't see a need for a config if so.

Reply:
Yeah, that is a good idea.
AFAIK, there is no downside and deferRender is a general optimization.
I will do it tomorrow, thanks.

Reply:
I have made deferRender by defaults for all the  expandable tables.